irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -247,19 +246,17 @@ namespace iroha {
       auto transactions_errors_log = validation::TransactionsErrors{};
       auto valid_proto_txs = validateTransactions(
           proposal.transactions(), temporaryWsv, transactions_errors_log, log_);
-      auto validated_proposal = shared_model::proto::ProposalBuilder()
-                                    .createdTime(proposal.createdTime())
-                                    .height(proposal.height())
-                                    .transactions(valid_proto_txs)
-                                    .createdTime(proposal.createdTime())
-                                    .build();
+
+      // Since proposal came from ordering gate it was already validated.
+      // All transactions has been validated as well
+      // This allows for unsafe construction of proposal
+      auto validated_proposal = factory_->unsafeCreateProposal(
+          proposal.height(), proposal.createdTime(), valid_proto_txs);
```

Comment:
Won't there be a performance increase, if we moved the ```valid_proto_txs```?

Reply:
unsafeCreateProposal accepts boost range, so no copy of vector will actually happen